src/pybind/mgr/telemetry/module.py

```diff
@@ -18,6 +18,33 @@
 
 ALL_CHANNELS = ['basic', 'ident', 'crash', 'device']
 
+# If the telemetry revision has changed since this point, re-nag the
+# user about opting in.  For example, we might do this every major
+# Ceph release.
+LAST_REVISION_NAG = 2
+
+# If the telemetry revision has changed since this point, re-require
+# an opt-in.  This should happen each time we add new information to
+# the telemetry report.
+LAST_REVISION_RE_OPT_IN = 2
+
+# Latest revision of the telemetry report.  Bump this each time we make
+# *any* change.
```

Comment:
I think we should only bump this if we materially/significantly increase the amount of information shared. e.g., requiring the user to opt-in again after we've dropped a field or obfuscated one seems like it'd annoy users, similarly if we simply renamed one. Or, to pick a recent example, revamped how the version numbers (for which we already got their consent) are aggregated.
Basically, once we change policy, not every execution detail.

I'm fine with the mechanism in general, I just don't want our telemetry consent questions to be too frequent and so annoying that users just turn it off. That's something to consider early on, because we need their consent to that ;-)

Reply:
Agree.  I'm not sure there is a lot of value in REVISION beyond just documenting the changes below... LAST_REVISION_RE_OPT_IN is the one that really matters.